pkg/controller/master-controller-manager/cluster-template-synchronizer/controller.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clustertemplatesynchronizer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+
+	kubermaticapiv1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	controllerutil "k8c.io/kubermatic/v2/pkg/controller/util"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	kuberneteshelper "k8c.io/kubermatic/v2/pkg/kubernetes"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"k8s.io/client-go/tools/record"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	// This controller syncs the kubermatic cluster templates on the master cluster to the seed clusters.
+	ControllerName = "cluster_template_syncing_controller"
+)
+
+type reconciler struct {
+	log          *zap.SugaredLogger
+	masterClient ctrlruntimeclient.Client
+	seedClients  map[string]ctrlruntimeclient.Client
+	recorder     record.EventRecorder
+}
+
+func Add(
+	masterMgr manager.Manager,
+	seedManagers map[string]manager.Manager,
+	log *zap.SugaredLogger) error {
+
+	log = log.Named(ControllerName)
+	r := &reconciler{
+		log:          log,
+		masterClient: masterMgr.GetClient(),
+		seedClients:  map[string]ctrlruntimeclient.Client{},
+		recorder:     masterMgr.GetEventRecorderFor(ControllerName),
+	}
+
+	c, err := controller.New(ControllerName, masterMgr, controller.Options{
+		Reconciler: r,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to construct controller: %v", err)
```

Comment:
Can we replace `%v` with `%w` also in other places where we do `fmt.Errorf`? Go introduced `%w` in v1.13, and it allows to wrap/unwrap errors, and I think it is better in general cases to use `%w`.
Ref: https://blog.golang.org/go1.13-errors#TOC_3.3.